src/aws-code-deploy/orb.yml

```diff
@@ -0,0 +1,237 @@
+              echo "No deployment group named << parameters.deployment-group >> found. Trying to create a new one"
+              aws deploy create-deployment-group \
+                --application-name << parameters.application-name >> \
+                --deployment-group-name << parameters.deployment-group >> \
+                --deployment-config-name << parameters.deployment-config >> \
+                --service-role-arn << parameters.service-role-arn >>
+            else
+              set -e
+              echo "Deployment group named << parameters.deployment-group >> already exists. Skipping creation."
+            fi
+  push-bundle:
+    parameters:
+      application-name:
+        description:
+          "The name of an AWS CodeDeploy application associated with the applicable IAM user or AWS account."
+        type: string
+      bundle-source:
+        description:
+          "The directory relative to your project to package up into an application revision."
+        type: string
+        default: "."
+      bundle-bucket:
+        description:
+          "The s3 bucket where an application revision will be stored"
+        type: string
+      bundle-key:
+        description:
+          "A key under the s3 bucket where an application revision will be stored"
+        type: string
+      bundle-type:
+        description:
+          "The file type used for an application revision bundle. Currently defaults to 'zip'"
+        type: string
+        default: "zip"
+    steps:
+      - run:
+          name: push-bundle
+          command: |
+            aws deploy push \
+              --application-name << parameters.application-name >> \
+              --source << parameters.bundle-source >> \
+              --s3-location s3://<< parameters.bundle-bucket >>/<< parameters.bundle-key >>.<< parameters.bundle-type >>
+
+  deploy-bundle:
+    parameters:
+      application-name:
+        description:
+          "The name of an AWS CodeDeploy application associated with the applicable IAM user or AWS account."
+        type: string
+      deployment-group:
+        description:
+          "The name of a new deployment group for the specified application."
+        type: string
+      deployment-config:
+        description:
+          "Predefined deployment configuration name."
+        type: string
+        default: "CodeDeployDefault.OneAtATime"
+      bundle-bucket:
+        description:
+          "The s3 bucket where an application revision will be stored"
+        type: string
+      bundle-key:
+        description:
+          "A key under the s3 bucket where an application revision will be stored"
+        type: string
+      bundle-type:
+        description:
+          "The file type used for an application revision bundle. Currently defaults to 'zip'"
+        type: string
+        default: "zip"
+    steps:
+      - run:
+          name: deploy-bundle
+          command: |
+            ID=$(aws deploy create-deployment \
```

Comment:
I think you want `set -e` in this code block

Reply:
My understanding is that by default the command block will be wrapped with`-eo pipefail` .

Reply:
Yep, the [docs confirm](https://circleci.com/docs/2.0/configuration-reference/#default-shell-options) they are, disregard my last comment.